src/cloudimagedirectory/format/format_google.py

```diff
@@ -55,8 +56,18 @@ def image_rhel(image: dict[str, str]) -> dict[str, str]:
 
     name = " ".join([x for x in name_parts if x != ""])
 
+    selflink_list = selflink.split("/")
+    if len(selflink_list) < 7 or selflink_list[5] != "projects":
+        # NOTE: We extract the current project name from the original selflink.
+        # Example selflinks:
+        # https://.../compute/imagesDetail/projects/rhel-cloud/.../...
+        # https://.../compute/imagesDetail/projects/rhel-sap-cloud/.../...
+        raise Exception("invalid selflink")
+
+    project_name = selflink.split("/")[6]
+
     selflink = "https://console.cloud.google.com/compute/imagesDetail/"
```

Comment:
Out of curiosity, why don't you just keep the original selflink? It looks like the front-end is using your selflink for the "Launch now" button, but then [recreates the original selflink](https://github.com/redhatcloudx/cloud-image-directory-frontend/blob/26177ac85e27026ce0b80bfb0e13257309ff0bc1/src/app/components/DetailsView.tsx#L102-L108
) for the "Launch from CLI" section. That doesn't look optimal.

Also, the selflink is really important for the actual identification of the image. The image id is not a reliable identifier, so I don't recommend to mess with this attribute.